### PR TITLE
replace pylint --> flake8

### PR DESCRIPTION
**Pull request by ilnarkz**
[#487] replace pylint --> flake8

### CODE REVIEW

**Comment by ilnarkz** [comment]
I didn't fix the errors. I didn't ignore anything.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+ignore =
```
Comment (elisno) [inline setup.cfg:2]: I know this goes against the suggestion in the issue mentioned in the description, but for this PR I think it's easiest to just focus on the errors `F401,F403` (which the CI workflow is supposed to handle).

Try adding this @ilnarkz:
```suggestion
ignore =
select = F401,F403
per-file-ignores =
    cleanlab/__init__.py: F401
    cleanlab/token_classification/__init__.py: F401
    cleanlab/benchmarking/__init__.py: F401
```

Comment (elisno) [reply setup.cfg:2]: The .pylintrc config had the setting:
https://github.com/cleanlab/cleanlab/blob/886e6aee6e81df463afbf9ff6e55a7d489343a39/.pylintrc#L408-L409

which doesn't exactly have a flake8-counterpart.

Comment (ilnarkz) [reply setup.cfg:2]: @elisno But then flake8 will only track these errors. Is this correct?

Comment (elisno) [reply setup.cfg:2]: You're right. That's the only thing at the moment that pylint is doing in our CI.

To keep the scope of this PR manageable, you should aim to replace that functionality without adding anything extra.

We can always open a new PR later for tracking more (or all) errors with flake8.

**Comment by elisno** [comment]
@ilnarkz Can you fill out the CLA linked above?

**Comment by ilnarkz** [comment]
@elisno  Ok, I did it.

**Pull request merged**